drivers/SmartThings/matter-appliance/src/matter-refrigerator/init.lua

```diff
@@ -15,29 +15,36 @@
 local capabilities = require "st.capabilities"
 local clusters = require "st.matter.clusters"
 local embedded_cluster_utils = require "embedded-cluster-utils"
-
 local log = require "log"
 local utils = require "st.utils"
-
 local version = require "version"
+
+local REFRIGERATOR_DEVICE_TYPE_ID = 0x0070
+local TEMPERATURE_CONTROLLED_CABINET_DEVICE_TYPE_ID = 0x0071
+
 if version.api < 10 then
   clusters.RefrigeratorAlarm = require "RefrigeratorAlarm"
   clusters.RefrigeratorAndTemperatureControlledCabinetMode = require "RefrigeratorAndTemperatureControlledCabinetMode"
   clusters.TemperatureControl = require "TemperatureControl"
 end
 
-local REFRIGERATOR_DEVICE_TYPE_ID = 0x0070
-local TEMPERATURE_CONTROLLED_CABINET_DEVICE_TYPE_ID = 0x0071
+local COMPONENT_TO_ENDPOINT_MAP = "__component_to_endpoint_map"
+local SUPPORTED_TEMPERATURE_LEVELS_MAP = "__supported_temperature_levels_map"
+local SUPPORTED_REFRIGERATOR_TCC_MODES_MAP = "__supported_refrigerator_tcc_modes_map"
+
+-- This is a work around to handle when units for temperatureSetpoint is changed for the App.
+-- When units are switched, we will never know the units of the received command value as the arguments don't contain the unit.
+-- So to handle this we assume the following ranges considering usual laundry temperatures:
+--   1. if the received setpoint command value is in range -6 ~ 20, it is inferred as *C
```

Comment:
I think this range would work well for refrigerator, but I think we'd need to be able to have the range go down to 0 degF/-18 degC (give or take a few degrees) for freezers. 0 degF is the USDA recommended temp for a freezer: https://ask.usda.gov/s/article/What-is-the-right-temperature-for-the-freezer. 

Using the component to endpoint map and vice versa, you can determine if the setpoint value is received from the freezer component and then adjust the values from there.

Reply:
I originally implemented in this way, but then looked at the `refrigerator-freezer-tn-tl` profile and saw that the freezer did not include the `temperatureSetpoint` capability. I assumed this was the case for all the profiles so I only implemented the temperature range limits for the refrigerator. But, it looks like `temperatureSetpoint` does exist in 2/4 of the refrigerator profiles, so I will make the change you suggested. It's clear why it is not in `refrigerator-freezer-tl.yml`, but any idea why it is not in `refrigerator-freezer-tn-tl.yml`?

Reply:
Implemented this in b6c2b0853c580cc6ae4bedde861a8b78d7b8e274!